File: db4oj/core/src/com/db4o/MTaIsDeleted.java
/* Copyright (C) 2004   db4objects Inc.   http://www.db4o.com */

package com.db4o;

import com.db4o.foundation.network.YapSocket;

final class MTaIsDeleted extends MsgD {
	public MTaIsDeleted() {
		super();
	}

	public MTaIsDeleted(MsgCloneMarker marker) {
		super(marker);
	}

	final boolean processMessageAtServer(YapSocket sock) {
	    YapStream stream = getStream();
		synchronized (stream.i_lock) {
			boolean isDeleted = getTransaction().isDeleted(this.readInt());
			int ret = isDeleted ? 1 : 0;
			Msg.TA_IS_DELETED.getWriterForInt(getTransaction(), ret).write(stream, sock);
		}
		return true;
	}
	
	public Object shallowClone() {
		return shallowCloneInternal(new MTaIsDeleted(MsgCloneMarker.INSTANCE));
	}
}

File: db4on/core/src/core/com/db4o/MTaIsDeleted.cs
namespace com.db4o
{
	internal sealed class MTaIsDeleted : com.db4o.MsgD
	{
		internal sealed override bool ProcessMessageAtServer(com.db4o.foundation.network.YapSocket
			 sock)
		{
			com.db4o.YapStream stream = GetStream();
			lock (stream.i_lock)
			{
				bool isDeleted = GetTransaction().IsDeleted(this.ReadInt());
				int ret = isDeleted ? 1 : 0;
				com.db4o.Msg.TA_IS_DELETED.GetWriterForInt(GetTransaction(), ret).Write(stream, sock
					);
			}
			return true;
		}
	}
}
